Update FAQ with chunking and virtualization details (#1007)
zarr-developers/VirtualiZarr: docs/faq.md

VirtualiZarr works by mapping your data to the zarr data model from whatever data model is used by the format it was saved in.

> This means that if your data contains anything that cannot be represented within the zarr data model, it cannot be virtualized.

Reviewer: ```suggestion
This means that if your data contains anything that cannot be represented within the zarr data model, it cannot be virtualized. The following restrictions influence whether you can virtualize a data file.
```

I think a pre-ample to the list would help

> - **Format chunks span contiguous byte ranges** - It's only possible to efficiently access individual chunks of data inside blobs in object storage if each chunk can be fetched via a single HTTP range request, which requires each chunk to occupy a contiguous localized series of bytes within the file layout. Well-designed formats such as netCDF and GRIB have this property, but other formats such as CSV do not. Note also this means that any additional processing which scrambles the byte locations will prevent virtualization - a single netCDF file is virtualizable, but a zipped or gzipped netCDF file is not!

Reviewer: ```suggestion
- **File must contain chunks of data, where each chunk spans a contiguous segment of the file** - For virtualization to work, each chunk must occupy a contiguous localized series of bytes within the file layout, so that the chunks can be fetched via a single HTTP range request. Well-designed formats such as netCDF and GRIB have this property, but other formats such as CSV do not. Note also this means that any additional processing which scrambles the byte locations will prevent virtualization - a single netCDF file is virtualizable, but a zipped or gzipped netCDF file is not!
```
